### DISCUSSION TITLE
iPXE booting Rocky 9 from Repo (UEFI)

### DISCUSSION THREAD

**Discussion by jamaya77** [ipxe/ipxe#1017] [Q&A] [answered]
Hi,
This isn't issue with ipxe itself, in fact its been working great. I was just curious if anyone in the forum has tried to setup Rocky Linux 9 to boot from repo via ipxe, more specifically for uefi machines?

This is my current configuration which seems to working fine with legacy (or bios) machines, but it won't work with uefi machines.

**

#!ipxe

set base http://download.rockylinux.org/pub/rocky/9/BaseOS/x86_64/os/images/pxeboot
set repo http://download.rockylinux.org/pub/rocky/9/BaseOS/x86_64/os

set params inst.text

imgfree
kernel ${base}/vmlinuz ip=dhcp inst.repo=${repo} ${params} ${ipparam} initrd=initrd
initrd ${base}/initrd.img
boot

**

**Comment by NiKiZe** [comment]
If this is older version of iPXE, or older kernel, you need to have a initrd= on your cmdline. You have initrd=initrd, but your initrd is actually named initrd.img so adding the missing .img would hopefully make this work.

**Reply by NiKiZe** [reply, marked as answer]
Please reply in the correct location
See https://github.com/ipxe/ipxe/commit/1192e from 2001
But go with fixing the initrd= part first.

After that upgrade iPXE and see that everything still works. And finally remove initrd= and see if it works as well.
As it might not work with older versions of the Linux kernel, and you didn't post which one is used there.

**Reply by jamaya77** [reply]
Sorry, this is the current kernel using on the server (Centos 8):

[root@ipxe ~]# uname -r
4.18.0-348.2.1.el8_5.x86_64

**Reply by jamaya77** [reply]
On second thought, I might consider upgrading to Rocky Linux 9 and just migrate over and start fresh. Any recommendation (or documentation) as to how the best approach? Is Rocky supported as a server for iPXE?

**Reply by jamaya77** [reply]
NikiZe,

Thank you for your help. Your suggestion worked like a charm! I upgraded iPXE and also used this opportunity to migrate the server to Rocky. After removing "initrd=", the script worked with both uefi and bios.  Thanks again. 


iPXE> show version
builtin/version:string = 1.21.1+ (gc1834)

**Reply by NiKiZe** [reply]
The question is in regards to which kernel version you are booting.

What you are hosting your boot services on is not relevant.

**Comment by jamaya77** [comment]
Hi Nikize,

The version of ipxe I'm using is "builtin/version:string = 1.21.1+ (g1192e)".  Not sure how far behind I am, but if its too old, could you point me in the right direction? It's been a while since I set this up. There's two files I'm aware I need to create (ipxe.efi and undionly.kpxe) that are configured in dnsmasq.

Thanks in advance.
